stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -687,8 +689,9 @@ bool swift::_checkGenericRequirements(
     case GenericRequirementKind::BaseClass: {
       // Demangle the base type under the given substitutions.
       auto baseType =
-        swift_getTypeByMangledName(req.getMangledTypeName(), substGenericParam,
-                                   substWitnessTable);
+        swift_getTypeByMangledName(MetadataState::Complete,
+                                   req.getMangledTypeName(), substGenericParam,
+                                   substWitnessTable).getMetadata();
```

Comment:
I notice you ask for `Complete` rather than `Abstract` metadata here. Is there a reason the metadata for a base class constraint would be guaranteed to be complete more than for a same type constraint?

Reply:
No, I think that was a mistake.  Most of the other places here were followed by `checkMetadataState` calls, but this wasn't.

Reply:
I've fixed this in the runtime-only PR, #22386.